### PR TITLE
[#164829364] Adds Create, Read, Update and Delete endpoints for articles app

### PR DESCRIPTION
**Pull request by Raywire**
**What does this PR do?**
Adds endpoints to Create, Read, Update, Delete articles and upload images.

**Description of the task to be completed**
A user should be able to create, read, update and delete an article.
A user should be able to upload images and retrieve them.

How should this be manually tested?
Once you clone the repo, open your terminal window and navigate to the root directory.
Create a virtual environment by running virtualenv venv and then activate the environment by
typing 
`source venv/bin/activate`
Install all the required dependencies using pip by running 
`pip install -r requirements.txt`
Run the Django server to serve up the app by typing 
`python manage.py runserver`
Using postman, send a request with the data described in the format below.

**Create an article** (**POST** localhost/api/articles/)
`{
    "article":
        {
            "title": "Authors Haven Test Four",
            "body": "Lorem ipsum dolor sit amet",
            "description": "Authors Haven Description Four",
            "is_published": true,
            "author": 2
        }
}
`
**Update an article** (**PUT** localhost/api/articles/authors-haven-test)
`{
    "article":
        {
            "title": "Authors Haven Test Updated",
            "body": "Lorem ipsum dolor sit amet",
            "is_published": true,
            "author": 2
        }
}`

**Get all articles**
**GET** `localhost/api/articles/`

**Get a specific article**
**GET** `localhost/api/articles/authors-haven-test/`

**Delete a specific article**
**DELETE** `localhost/api/articles/authors-haven-test/`

**Upload an image**
**POST** `localhost/api/articles/authors-haven-test/image/`
Using Postman, 

- Select the Body tab
- Ensure the parameter option is set to form-data
- Under the KEY field add the word file
- When you hover on the far right side of the KEY field a dropdown will appear. Select File among the options in the dropdown
-  In the VALUE field, press the Select Files button and choose an image file on the file picker that pops up.

**Retrieve images of an article**
This retrieves all images associated with the article slug
**GET** `localhost/api/articles/authors-haven-test/image/`

**Relevant PT stories**
https://www.pivotaltracker.com/story/show/164829364

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from .models import Article
+
+
+class ArticleSerializer(serializers.ModelSerializer):
+    """Serializer to map the Article Model instance into JSON format."""
+
+    title = serializers.CharField(max_length=100)
+    description = serializers.CharField(max_length=128)
+    body = serializers.CharField()
+    # author = serializers.IntegerField()
```
Comment (WNjihia) [inline authors/apps/articles/serializers.py]: kindly remove the commented code

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    def create(self, validated_data):
+        return Article.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        instance.title = validated_data.get('title', instance.title)
+        instance.description = validated_data.get(
+            'description', instance.description)
+        instance.body = validated_data.get('body', instance.body)
+        # instance.author_id = validated_data.get('author_id', instance.author_id)
```
Comment (WNjihia) [inline authors/apps/articles/serializers.py]: Same goes for this. kindly remove the commented out

File: authors/apps/articles/serializers.py (hunk 1 above)
Comment (Raywire) [reply authors/apps/articles/serializers.py]: Done thanks for the feedback.

File: authors/apps/articles/serializers.py (hunk 2 above)
Comment (Raywire) [reply authors/apps/articles/serializers.py]: Also done, the feedback is appreciated.

**Review by kihuha — DISMISSED** [review]

**Review by kihuha — APPROVED** [review]

**Review by WNjihia — APPROVED** [review]

**Pull request merged**